cluster-api/src/test/java/io/aiven/klaw/clusterapi/services/ApacheKafkaTopicServiceTest.java

```diff
@@ -0,0 +1,461 @@
+
+    Mockito.when(clusterApiUtils.getAdminClient(environment, protocol, clusterIdentification))
+        .thenReturn(adminClient);
+    Mockito.when(adminClient.listTopics(any(ListTopicsOptions.class))).thenReturn(listTopicsResult);
+    Mockito.when(listTopicsResult.names()).thenReturn(KafkaFuture.completedFuture(Set.of("name")));
+    Mockito.when(adminClient.describeTopics(any(Collection.class)))
+        .thenReturn(describeTopicsResult);
+    Mockito.when(describeTopicsResult.all())
+        .thenReturn(KafkaFuture.completedFuture(topicDescriptionsPerAdminClient));
+    Mockito.when(topicDescription.partitions()).thenReturn(List.of(topicPartitionInfo));
+
+    Set<TopicConfig> topicConfigs =
+        apacheKafkaTopicService.loadTopics(environment, protocol, clusterIdentification);
+
+    Assertions.assertThat(topicConfigs.size()).isEqualTo(1);
+  }
+
+  @Test
+  void createTopicClientNull() throws Exception {
+    String environment = "ENVIRONMENT";
+    String clusterIdentification = "CLUSTER_IDENTIFICATION";
+    ClusterTopicRequest clusterTopicRequest =
+        ClusterTopicRequest.builder()
+            .env(environment)
+            .clusterName(clusterIdentification)
+            .protocol(protocol)
+            .build();
+
+    Mockito.when(clusterApiUtils.getAdminClient(environment, protocol, clusterIdentification))
+        .thenReturn(null);
+
+    AbstractThrowableAssert<?, ? extends Throwable> exception =
+        assertThatThrownBy(() -> apacheKafkaTopicService.createTopic(clusterTopicRequest));
+
+    exception.isInstanceOf(Exception.class);
+    exception.hasMessage("Cannot connect to cluster.");
+  }
+
+  @Test
+  void createTopic() throws Exception {
+    String environment = "ENVIRONMENT";
+    String clusterIdentification = "CLUSTER_IDENTIFICATION";
+    String topicName = "TOPIC_NAME";
+    int partitions = 5;
+    short replicationFactor = 1;
+    Map<String, String> advancedTopicConfiguration = Map.of("topic", "config");
+    ClusterTopicRequest clusterTopicRequest =
+        ClusterTopicRequest.builder()
+            .env(environment)
+            .clusterName(clusterIdentification)
+            .protocol(protocol)
+            .topicName(topicName)
+            .partitions(partitions)
+            .replicationFactor(replicationFactor)
+            .advancedTopicConfiguration(advancedTopicConfiguration)
+            .build();
+
+    Mockito.when(clusterApiUtils.getAdminClient(environment, protocol, clusterIdentification))
+        .thenReturn(adminClient);
+    Mockito.when(adminClient.createTopics(anyCollection())).thenReturn(createTopicsResult);
+    Mockito.when(createTopicsResult.values())
+        .thenReturn(Map.of(topicName, KafkaFuture.completedFuture(null)));
+
+    ApiResponse response = apacheKafkaTopicService.createTopic(clusterTopicRequest);
+
+    Assertions.assertThat(response.isSuccess()).isTrue();
+    Assertions.assertThat(response.getMessage()).isEqualTo("success");
```

Comment:
You can change success to ApiResultStatus.SUCCESS.value to reference the ENUM

Reply:
I will make these changes.